### DISCUSSION TITLE
Why do I get this prompt every time I install on Windows?

### DISCUSSION THREAD

**Discussion by Yukixieyuya821** [tauri-apps/tauri#8046] [Q&A] [answered]
Microsoft Defender SmartScreen blocks unrecognized apps from launching. Running this application may put your computer at risk.

Application:
xxx_x64_en-US.msi
Publisher:
Published by unknown

Is there any way to solve it?

**Comment by FabianLars** [comment, marked as answer]
Only by [code signing](https://tauri.app/v1/guides/distribution/sign-windows) your app with an EV certificate (or an _established_ OV certificate - meaning a certificate that already saw a few thousands of downloads).
The problem with EV certificates is that only companies can get it :/ Also, both EV and OV certificates are stubidly expensive.

P.S. One thing i can't remember is if you can make the prompt at least show a name instead of "unknown" if you use a self-signed certificate 🤔

**Reply by Yukixieyuya821** [reply]
I'm not using a self-signed certificate. 
I understand🤔, thanks for your reply

**Comment by mnaza** [comment]
Since this is still the top result for the question, an update for anyone landing here in 2026.

Fabian's answer was right when it was written. Since 2024 it is not: Microsoft removed the special treatment of EV code-signing certificates from its Trusted Root Program (EV code-signing certificates no longer recognised from February 2024, EV OIDs removed from roots in August 2024), so EV and OV now build SmartScreen reputation the same way and neither skips the prompt on a new release. The Tauri signing guide reflects this now: https://v2.tauri.app/distribute/sign/windows/

What the prompt actually keys on is reputation, and reputation is attached to the file's hash and to the signing certificate (https://learn.microsoft.com/en-us/windows/apps/package-and-deploy/smartscreen-reputation). Every new build starts at zero on the hash side; the certificate is the only thing that carries over, and only if you keep signing with the same one. So for a Tauri app the practical answer is: sign every release with one certificate and keep it, expect the prompt on early downloads of each release, and do not buy EV for this. The one thing that removes the prompt entirely is Microsoft Store distribution, where Microsoft re-signs the package.

On the P.S.: a self-signed certificate does not change the prompt at all; Microsoft's table lists it as "same behavior as no signature". Only a certificate chaining to a CA in the Trusted Root Program replaces "unknown publisher" with your name.

Fuller write-up with Microsoft's wording on what resets reputation: https://signalscreen.dev/blog/smartscreen-warning-signed-installer?utm_source=github&utm_campaign=tauri-discussion-8046
